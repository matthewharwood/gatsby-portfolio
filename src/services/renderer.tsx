function componentMapFactory() {
  const componentMap = {
    "postHeader": (props: any) => console.log("POSTHEADER", props),
    "summary": (props: any) => console.log("SUMMARY", props),
  }

  let defaultValueHandler = {
    get: function(target: any, name: any) {
      return target.hasOwnProperty(name)
        ? target[name]
        : (props: any) => console.warn(
          "We are missing a component type: ", props._type, "is not found"
        );
    },
  }

  return new Proxy(componentMap, defaultValueHandler)
}

export const pageRenderer = (raw) => {

  const componentMap = componentMapFactory();

  raw.map((d) => [d._type, d]).map(([key, props]) => componentMap[key](props));
  return true
}