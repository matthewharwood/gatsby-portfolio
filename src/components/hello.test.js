import React from "react"
import renderer from "react-test-renderer"
import Hello from './hello';

describe("Hello", () => {
  it("renders correctly", () => {
    const tree = renderer
      .create(<Hello />)
      .toJSON()
    expect(tree).toMatchSnapshot();
  })
})