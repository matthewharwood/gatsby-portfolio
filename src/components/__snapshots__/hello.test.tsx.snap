// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hello renders correctly 1`] = `
<div>
  hello
</div>
`;
